requirements-dev.txt

```diff
@@ -3,7 +3,7 @@ astroid==2.9.3
 pylint==2.12.2
 black~=22.0
 qiskit-sphinx-theme>=1.6
-sphinx-autodoc-typehints
+sphinx-autodoc-typehints~=1.18,!=1.19.3
```

Comment:
Without knowing too much about this particular package, I'd prefer the simpler `sphinx-autodoc-typehints>=1.19.4` here simply because for dev requirements there is no harm in being picky (as environments should always be custom for building docs).

Reply:
just implemented this - seems fine locally, hopefully things build